### PR TITLE
Added description for local registries (and air gapped application install)

### LINKED ISSUE #26: Local Registries (and Air Gapped application install)

### PR DESCRIPTION
**Pull request by arne-broering**
### Description

Added the file `local-registries.md` to explain the usage and purpose of local registries.

#### Issues Addressed

#26 

#### Change Type

Please select the relevant options:

- [ ] Fix (change that resolves an issue)
- [x] New enhancement (change that adds specification content)

#### Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) document.
- [x] My changes adhere to the established patterns, and best practices.

### CODE REVIEW

File: system-design/app-interoperability/local-registries.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Local Registries
+
+This section investigates options for configuring the usage of local Docker (or Helm Chart) registries. The goal of configuring such local registries is to avoid the reliance on public, Internet-accessible registries. The reasons for not using such public registries are mainly twofold: (1) publicly hosted Docker images or Helm charts could become unavailable at some point, as the owner decides to take the Docker images or Helm charts off the public registry, or (2) Internet connectivity may not be available to the device and hence public registries are not reachable.
+
```
Comment (phil-abb) [inline system-design/app-interoperability/local-registries.md:4]: I think a third point would be that end-users want to host their own registries so they can do security scans and validate the packages

File: system-design/app-interoperability/local-registries.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+configs:
+  "docker.io":
+    auth:
+      username: "<username>"
+      password: "<password>"
+```
+
+## Option - Docker Registry as Pull-through Cache on Docker Level
+
+To configure a pull-through cache in Docker for the container registry, a Docker Registry can be setup that acts as caching proxy for a remote Docker registry Such a Docker Registry container can be defined using the following `config.yml`:
```
Comment (phil-abb) [inline system-design/app-interoperability/local-registries.md]: Maybe missing a period between "remote Docker registry <period> Such as Docker Registry..."

File: system-design/app-interoperability/local-registries.md (hunk 1 above)
Comment (arne-broering) [reply system-design/app-interoperability/local-registries.md:4]: yes, I added this point to the text and you find it in the latest commit.

File: system-design/app-interoperability/local-registries.md (hunk 2 above)
Comment (arne-broering) [reply system-design/app-interoperability/local-registries.md]: yes, I just added it.

**Review by phil-abb — APPROVED** [review]

**Review by ajcraig — APPROVED** [review]
Thanks for adding the content Arne. I believe the local vs. remote repositories and configuration will eventually end up in a device orchestration/configuration location but good to get it added.

**Pull request merged**